sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -887,8 +1125,16 @@ size_t gcoap_req_send(const uint8_t *buf, size_t len,
         }
     }
 
+    ssize_t res = 0;
+    coap_socket_t socket = { 0 };
+
+    _tl_get_socket(&socket);
+#if IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS)
```

Comment:
You can drop the `#ifdef` with the change suggested above

Reply:
Dropped, and checked  the socket type here.